certification/internal/authn/keychain.go

```diff
@@ -0,0 +1,81 @@
+package authn
+
+import (
+	"os"
+	"sync"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/types"
+	craneauthn "github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+type preflightKeychain struct {
+	mu sync.Mutex
+}
+
+var PreflightKeychain craneauthn.Keychain = &preflightKeychain{}
```

Comment:
Should we stuffing the authentication context into `context` using `WithValue()?`, which is already getting passed around everywhere? As opposed to using a global here?

Reply:
I actually had that thought. I think it probably is the best way to handle it. If certain commands require different keychains, it would simplify it.

Reply:
Let's punt on this one for a more comprehensive PR on just this subject.